### ISSUE TITLE
problem with train.py in python3, IndexError: index 2 is out of bounds for dimension 0 with size 1

### ISSUE THREAD

**Issue by vishnu41** [andy-yun/pytorch-0.4-yolov3#57]
I'm getting this error only while using python3
![Screenshot (32)](https://user-images.githubusercontent.com/40159472/54676064-da4f1600-4b25-11e9-9ad3-65e1c0c35f73.png)

find attached cfg file
[yolo-voc.txt](https://github.com/andy-yun/pytorch-0.4-yolov3/files/2987589/yolo-voc.txt)

**Comment by andy-yun** [comment]
Attached file is used at yolov2. But your error is occurred while yolov3 is running. 
Would you check the command or cfg once more?

**Comment by vishnu41** [comment]
Got it. The problem was with the data. Anyway thanks for the code man

**Comment by mrkieumy** [comment]
Hi @andy-yun , I meet the same this error in test function. 
Training is normal, no error. But:
Because I want to see the test loss, so I copy 3 lines in train function to test function:
test_loss = 0.0
for i, l in enumerate(loss_layers):
       ol = l(output[i]['x'], target)
       test_loss += ol.item()
And when run to this line in test function ( ol = l(output[i]['x'], target) ) , the error is: 
.........
File "/home/kieumy/YOLO/pytorch/yolo_layer.py", line 152, in forward
    self.build_targets(pred_boxes, target.detach(), anchors.detach(), nA, nH, nW)
  File "/home/kieumy/YOLO/pytorch_conditioning/yolo_layer.py", line 102, in build_targets
    tcls[b][best_n][gj][gi][int(tbox[t][0])] = 1
IndexError: index 1 is out of bounds for dimension 0 with size 1
------------------------------------
I saw on the code of build_targets that the index best_n is a torch tensor (0), so I tried best_n = int(best_n.item()). But the error's still there.
I don't understand because, during training, it runs normally, but when I call test function, this error occurs.
Does anyone know the reason why?
Thanks.
